### ISSUE TITLE
Semgrep 1.5.1 release is broken on macOS

### ISSUE THREAD

**Issue by det** [semgrep/semgrep#6904]
**Describe the bug**
In semgrep 1.5.1, but not previous versions, `semgrep-core` is dynamically linked against a library that no user will have on their system.

```
$ otool -L semgrep-core
semgrep-core:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/Users/ssm-user/actions-runner/_work/semgrep/semgrep/libs/ocaml-tree-sitter-core/tree-sitter/lib/libtree-sitter.0.dylib (compatibility version 0.0.0, current version 0.0.0)
```

**What is the priority of the bug to you?**

- [X] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [ ] P2: regular bug that should get fixed

**Environment**
macOS x86_64

**Comment by emjin** [comment]
Hi! Can you share how you installed semgrep?

**Comment by det** [comment]
`pip install semgrep`

**Comment by ikonst** [comment]
Seeing this too:
```
$ semgrep --version
1.5.1
$ semgrep debug
...
Running Semgrep engine with command:
.../lib/python3.8/site-packages/semgrep/bin/semgrep-core -json -rules ...
$ .../lib/python3.8/site-packages/semgrep/bin/semgrep-core -json -rules ...
dyld[71111]: Library not loaded: /Users/ssm-user/actions-runner/_work/semgrep/semgrep/libs/ocaml-tree-sitter-core/tree-sitter/lib/libtree-sitter.0.dylib
  Referenced from: <B31139C5-3FC3-34E7-8042-2CCE242BD279> .../lib/python3.8/site-packages/semgrep/bin/semgrep-core
  Reason: tried: '/Users/ssm-user/actions-runner/_work/semgrep/semgrep/libs/ocaml-tree-sitter-core/tree-sitter/lib/libtree-sitter.0.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/ssm-user/actions-runner/_work/semgrep/semgrep/libs/ocaml-tree-sitter-core/tree-sitter/lib/libtree-sitter.0.dylib' (no such file), '/Users/ssm-user/actions-runner/_work/semgrep/semgrep/libs/ocaml-tree-sitter-core/tree-sitter/lib/libtree-sitter.0.dylib' (no such file), '/usr/local/lib/libtree-sitter.0.dylib' (no such file), '/usr/lib/libtree-sitter.0.dylib' (no such file, not in dyld cache)
zsh: abort       -json -rules  -j 10 -targets  -timeout 30 -timeout_threshold 3 -max_memory 0
```

Does not reproduce with semgrep 1.3.0 (`pip install semgrep==1.3.0`).

**Comment by brandonspark** [comment]
@mjambon Any idea on why this might be occurring? Emma and I are having trouble reproducing

**Comment by emjin** [comment]
@det @ikonst Thanks so much for filing this issue! We think we've identified the problem, in which case this will be fixed in the next release. I'll update when we have more confirmed

**Comment by emjin** [comment]
We've confirmed the fix works, so this should be resolved next release, schedule for tomorrow.

**Comment by emjin** [comment]
Fix released! Update with `pip install semgrep==1.6.0`. I'll close this now; please open another issue if you continue to have problems

**Comment by det** [comment]
Thank you!
